html/includes/api_functions.inc.php

```diff
@@ -2027,6 +2027,42 @@ function get_link()
 }
 
 
+function list_sensors()
+{
+    $app        = \Slim\Slim::getInstance();
+    $router     = $app->router()->getCurrentRoute()->getParams();
+    $sql        = '';
+    $sql_params = array();
+    $hostname   = $router['hostname'];
+    $device_id  = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+    if (is_numeric($device_id)) {
+        check_device_permission($device_id);
+        $sql        = " AND `sensors`.`device_id`=?";
+        $sql_params = array($device_id);
+    }
+    if (!LegacyAuth::user()->hasGlobalRead()) {
+        $sql .= " AND `sensors`.`device_id` IN (SELECT device_id FROM devices_perms WHERE user_id = ?)";
+        $sql_params[] = LegacyAuth::id();
+    }
+
+    $sensors       = array();
+    foreach (dbFetchRows("SELECT `sensors`.* FROM `sensors` LEFT JOIN `devices` ON `sensors`.`device_id` = `devices`.`device_id` WHERE `sensors`.`sensor_id` IS NOT NULL $sql", $sql_params) as $sensor) {
+        $host_id = get_vm_parent_id($device);
```

Comment:
$device is not defined yet...

Reply:
Yes, missed that. Now i see that i wrongly duplicated this construct, it looks devices table specific with it's ip and parent_id stuff. Maybe to just use check_is_read for such list functions?